quetz/main.py

```diff
@@ -937,7 +947,12 @@ def post_channel_member(
         channel_member.role = new_member.role
         db.commit()
     else:
-        dao.create_channel_member(channel.name, new_member)
+        try:
+            dao.create_channel_member(channel.name, new_member)
+        except KeyError as error:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail=error.args[0]
+            ) from error
```

Comment:
Thinking about it: This changes user-facing behavior. Maybe let's put this in a separate PR.